keras_cv/layers/preprocessing/random_sharpness.py

```diff
@@ -0,0 +1,121 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class RandomSharpness(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Randomly performs the sharpness operation on given images.
+
+    The sharpness operation first performs a blur operation, then blends between the
+    original image and the blurred image.  This operation makes the edges of an image
+    less sharp than they were in the original image.
+
+    References:
+        [Sharpness](https://pillow.readthedocs.io/en/stable/reference/ImageEnhance.html)
+
+    Args:
+        factor: Either a tuple of two floats or a single float. `factor` controls the
+            extent to which the image sharpness is impacted.  `factor=0.0` makes this
+            layer perform a no-op operation, while a value of 1.0 uses the sharpened
+            result entirely.  Values between 0 and 1 result in linear interpolation
+            between the original image and the sharpened image.
+
+            Values should be between `0.0` and `1.0`.  If a tuple is used, a `factor` is
+            sampled between the two values for every image augmented.  If a single float
+            is used, a value between `0.0` and the passed float is sampled.  In order to
+            ensure the value is always the same, please pass a tuple with two identical
+            floats: `(0.5, 0.5)`.
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.  Defaults to
+            `[0, 255].`
+    """
+
+    def __init__(
+        self,
+        factor,
+        value_range=(0, 255),
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.value_range = value_range
+        self.factor = RandomSharpness._parse_factor(factor)
+
+    @staticmethod
+    def _parse_factor(factor):
+        if isinstance(factor, float):
+            factor = (0.0, factor)
+
+        if factor[0] > factor[1]:
+            raise ValueError("`factor[0] > factor[1]`, `factor[0]` must be <= "
+                f"`factor[1]`.  Got `factor={factor}`")
+        if factor[0] < 0.0 or factor[1] > 1.0:
+            raise ValueError("`factor` should be inside of range [0.0, 1.0]. "
+                f"Got factor={factor}")
+
+        return factor
+
+    def get_random_tranformation(self):
```

Comment:
tyoi

Reply:
This is about to be fixed in https://github.com/keras-team/keras/pull/16278.